src/node_crypto.cc

```diff
@@ -3708,11 +3822,11 @@ void Hmac::HmacInit(const char* hash_type, const char* key, int key_len) {
   if (md == nullptr) {
     return env()->ThrowError("Unknown message digest");
   }
-  HMAC_CTX_init(&ctx_);
+  HMAC_CTX_reset(ctx_);
```

Comment:
This one is pretty much redundant with the one at line 3985. The state here is really confusing. Since you're paying for a heap-allocated `EVP_MD_CTX` anyway, simpler would be to merge the `initialised_` bit into whether `ctx_` is null. That is:

1. `HmacInit` has a `CHECK_EQ(nullptr, ctx_)`. It then sets `ctx_ = HMAC_CTX_new()` before `HMAC_Init_ex`. (TBH, OpenSSL didn't do a great job of cleaning up the API while they were opaquifying. `HMAC_CTX_new` and `HMAC_Init_ex` should have been the same operation.
2. All the `if (!initialised_)` checks before `if (!ctx_)`
2. In `HmacDigest`, rather than reset and clearing `initialised_`, do `HMAC_CTX_free(ctx_);` and `ctx_ = nullptr;`.